otel-agent/ecs-ec2/ecslogresourcedetectionprocessor/ecslog_resourcedetection_processor.go

```diff
@@ -0,0 +1,121 @@
+package ecslogresourcedetectionprocessor
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/coralogix/telemetry-shippers/otel-agent/ecs-ec2/ecslogresourcedetectionprocessor/internal/metadata"
+	"github.com/coralogix/telemetry-shippers/otel-agent/ecs-ec2/ecslogresourcedetectionprocessor/internal/utils"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/zap"
+)
+
+// global variable used to store metadata endpoints within package
+var ecsMetadataHandler metadataHandler
+
+func processLogsFunc(logger *zap.Logger, c *Config) processorhelper.ProcessLogsFunc {
+	return func(ctx context.Context, ld plog.Logs) (plog.Logs, error) {
+		for i := 0; i < ld.ResourceLogs().Len(); i++ {
+			rlog := ld.ResourceLogs().At(i)
+			containerID := getContainerId(&rlog)
+			logger.Debug("processing",
+				zap.String("container.id", containerID),
+				zap.String("processor", metadata.Type))
+
+			// check for ecs agent, Note that ecs agents do not have
+			// metadata endpoints
+			if ok, _ := isECSAgent(containerID); ok {
+				rlog.Resource().Attributes().PutStr("ecs.agent", "true")
+			}
+
+			endpoints, ok := ecsMetadataHandler.get(containerID)
+			if !ok || len(endpoints) == 0 {
+				logger.Debug("metadata not found",
+					zap.String("container.id", containerID),
+					zap.String("processor", metadata.Type))
+				return ld, nil
+			}
+
+			resp, err := http.Get(endpoints[0])
+			if err != nil {
+				return ld, err
+			}
+
+			var data map[string]interface{}
```

Comment:
Wanted to keep things relatively generic here since we were only interested in the Key/Value pairs, however, for better readability, I'll add a struct.